src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php

```diff
@@ -219,6 +219,11 @@ public function regenerate($destroy = false, $lifetime = null)
      */
     public function save()
     {
+        // In PHP <7.2 session_write_close() returns void if the session is not started.
+        // @see https://www.php.net/manual/function.session-write-close.php
+        if (!$this->started) {
```

Comment:
I thought about this approach as well. I hesitated to go this way because of the phpdoc on the interface:

https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/Session/Storage/SessionStorageInterface.php#L107

Not sure if this would not also be a BC break as someone is relying on this? :confused:

Reply:
Actually it seems this `RuntimeException` is only really valid for the mock storages?

https://github.com/symfony/symfony/pull/6362/files

Reply:
Can we catch that exception in Session::save() and convert it to a warning or deprecation error saying that the storage needs to handle an attempt to save a reset session on its own?  The behavior that you describe is what existed prior to this BC break in 3.4.24 for custom storage, prior to that they would have had to handle a reset session on their own anyway.